pkg/providers/instancetype/instancetype.go

```diff
@@ -175,34 +172,28 @@ func (p *Provider) createOfferings(ctx context.Context, instanceType *ec2.Instan
 	return offerings
 }
 
-func (p *Provider) getAvailabilityZones(ctx context.Context) (sets.Set[string], error) {
+func (p *Provider) getInstanceTypeZones(ctx context.Context, instanceTypeOfferings map[string]sets.Set[string]) sets.Set[string] {
 	// DO NOT REMOVE THIS LOCK ----------------------------------------------------------------------------
 	// We lock here so that multiple callers to getAvailabilityZones do not result in cache misses and multiple
 	// calls to EC2 when we could have just made one call.
 	// TODO @joinnis: This can be made more efficient by holding a Read lock and only obtaining the Write if not in cache
 	p.mu.Lock()
 	defer p.mu.Unlock()
 	if cached, ok := p.cache.Get(AvailabilityZonesCacheKey); ok {
```

Comment:
```suggestion
	if cached, ok := p.cache.Get(ZonesCacheKey); ok {
```

Should this still be called the `AvailabilityZonesCacheKey` if it isn't representing just AZs?